app/controllers/security_ruleset_controller.rb

```diff
@@ -0,0 +1,40 @@
+
+class SecurityRulesetController < ApplicationController
+  include RulesetValidations
+
+  before_action :validate_file_upload!, only: [:create]
+  before_action :clear_errors
+
+  def new
+  end
+
+  def create
+    crunch_hash = JSON.parse(crunch42_data)
+    @score = crunch_hash['score']
+    @openapi = crunch_hash['openapiState']
+    @counter = crunch_hash['issueCounter']
+    @criticality = crunch_hash['criticality']
+    @issues = crunch_hash['data']['issues'].map do |issue|
```

Comment:
This line has moved and our previous comments have been detached from it so I'll reopen it here. As from the pry output posted in those comments, it appears there is still some confusion here.

The way I've got my head around it is to look at the fake 32Crunch file that was posted in a previous version of this code base (fakecrunchresults.json [at the root off this app](https://github.com/co-cddo/api-spec-linter/blob/main/fakecrunchresults.json)). If you pass the code to a JSON prettifier (For example https://jsonformatter.org/) then the relevant section of code starts:

```json
"data": {
    "issueCounter": 60,
    "score": 43.01708333333328,
    "criticality": 4,
    "issues": {
      "v3-response-406": {
        "description": "406 response should be defined for all GET, POST, PUT, PATCH, and DELETE operations",
        "issues": [
          {
            "score": -0.3,
            "pointer": 28,
            "tooManyImpacted": false,
            "criticality": 3,
            "response": true
          },
          {
            "score": -0.23333333333333334,
            "pointer": 35,
            "tooManyImpacted": false,
            "criticality": 3,
            "response": true
          }
        ],
        "issueCounter": 2,
        "score": -0.5333333333333333,
        "criticality": 3,
        "tooManyError": false
      },
```
The **issues** element at data/issues is a hash (line 5 above, line 178 in the prettified output).

Each value of that hash is also a hash, and that hash confusingly also contains a key **issues** which contains an array (line 8 above, line 181 in the prettified output).

So line 17 here is iterating over the outer hash.

The output you posted from pry was one of the inner issues elements.

Reply:
Ah yes I see now, adding _key makes the element `issue` easier to deal with as well. Cheers